File: 1.1/src/cz/mg/vulkan/PFNvkCmdClearColorImage.java
package cz.mg.vulkan;

public class PFNvkCmdClearColorImage extends VkFunctionPointer {
    public PFNvkCmdClearColorImage() {
    }

    protected PFNvkCmdClearColorImage(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdClearColorImage(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdClearColorImage(long value) {
        setValue(value);
    }

    public PFNvkCmdClearColorImage(VkInstance instance) {
        super(instance, new VkString("vkCmdClearColorImage"));
    }

    public void call(VkCommandBuffer commandBuffer, VkImage image, int imageLayout, VkClearColorValue pColor, int rangeCount, VkImageSubresourceRange pRanges){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), image != null ? image.getVkAddress() : VkPointer.getNullAddressNative(), imageLayout, pColor != null ? pColor.getVkAddress() : VkPointer.NULL, rangeCount, pRanges != null ? pRanges.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, long image, int imageLayout, long pColor, int rangeCount, long pRanges);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdClearColorImage.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdClearColorImage_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong image, jint imageLayout, jlong pColor, jint rangeCount, jlong pRanges){
    (void)env;
    (void)jc;
    PFN_vkCmdClearColorImage f = (PFN_vkCmdClearColorImage)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((VkImage*)jniLongToPointer(image)),
        (VkImageLayout)imageLayout,
        ((VkClearColorValue*)jniLongToPointer(pColor)),
        (uint32_t)rangeCount,
        ((VkImageSubresourceRange*)jniLongToPointer(pRanges))
    );
}

